### PR TITLE
Part 2 - Implementation task: Alternative implementation for the Notify Me component and flows

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/0/1203846112242642/f

### Description
Added Notify Me component and modified the downloads and the AppTP flows to use it.
Added a dummy listener for Notify Me in component preview.

### Steps to test this PR

Downloads flow on Android 13 - notifications permissions granted
- [x] Fresh install from this branch.
- [x] Open the app and allow notifications permissions.
- [x] Go to Downloads screen from Settings.
- [x] Notice there is no "Notify Me" component, there's an empty view.

Downloads flow on Android 13 - notifications permissions NOT granted (scenario close button)
- [x] Fresh install from this branch.
- [x] Open the app and deny notifications permissions.
- [x] Go to Downloads screen from Settings.
- [x] Notice the "Notify Me" component at the top and the empty view related text is below.
- [x] Download a file.
- [x] Notice the "Notify Me" component at the top and then the file item.
- [x] Tap on the X to close the "Notify Me" component.
- [x] Notice the component disappears and you don't see it even if you close the screen and come back.
- [x] Notice that if you enable App TP you see the "Notify Me" component on the Device Shield screen.

Downloads flow on Android 13 - notifications permissions NOT granted (scenario notify me button + deny)
- [x] Fresh install from this branch.
- [x] Open the app and deny notifications permissions.
- [x] Go to Downloads screen from Settings.
- [x] Notice the "Notify Me" component at the top and the empty view related text is below.
- [x] Download a file.
- [x] Notice the "Notify Me" component at the top and then the file item.
- [x] Tap on the "Notify me" button.
- [x] Notice the permissions request dialog. Tap on deny.
- [x] Notice the "Notify Me" component is still on screen.
- [x] Tap on the "Notify me" button.
- [x] Notice you are sent to System Settings. Come back and notice the "Notify Me" component is still on screen.
- [x] Tap on the "Notify me" button and allow permissions from the system settings.
- [x] Notice the "Notify Me" component disappears.

Downloads flow on Android 13 - notifications permissions NOT granted (scenario notify me button + allow)
- [x] Fresh install from this branch.
- [x] Open the app and deny notifications permissions.
- [x] Go to Downloads screen from Settings.
- [x] Notice the "Notify Me" component at the top and the empty view related text is below.
- [x] Download a file.
- [x] Notice the "Notify Me" component at the top and then the file item.
- [x] Tap on the "Notify me" button.
- [x] Notice the permissions request dialog. Tap on allow.
- [x] Notice the "Notify Me" component disappears from the screen.
- [x] Go to system settings and deny notifications permissions.
- [x] Come back to the Downloads screen and notice the "Notify Me" component is displayed on screen.

Run the above scenarios for the AppTP flow.

Android 12 or below
- [x] Fresh install from this branch.
- [x] Open the app and notice there is no prompt to grant permissions for notifications.
- [x] Go to Downloads screen and notice there is no "Notify Me" component displayed.
- [x] Enable AppTP and notice there is no "Notify Me" component displayed on Device Shield screen.

### UI changes
| Downloads Notify Me + Empty  | Downloads Notify Me + Item |
| ------ | ----- |
|downloads_notify_me_empty|downloads_notify_me_item|

| AppTP  Enabled + Notify Me  | AppTP  Disabled + Notify Me |
| ------ | ----- |
|apptp_enabled_notify_me|apptp_disabled_notify_me|

### CODE REVIEW

File: app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModelTest.kt [hunk 1]
```diff
@@ -345,13 +341,17 @@ class DeviceShieldTrackerActivityViewModelTest {
         assertEquals(BannerState.OnboardingBanner, bannerState)
     }
 
-    private fun createInMemoryDb(): VpnDatabase {
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModelTest.kt:348]: Removed as it was unused.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt [hunk 2]
```diff
@@ -145,7 +145,7 @@ class DeviceShieldTrackerActivity :
         }
 
         binding.ctaManageViewAllApps.setClickListener {
-            viewModel.onViewEvent(DeviceShieldTrackerActivityViewModel.ViewEvent.LaunchTrackingProtectionExclusionListActivity)
+            viewModel.onViewEvent(ViewEvent.LaunchTrackingProtectionExclusionListActivity)
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt:148]: Updated import.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModel.kt [hunk 3]
```diff
@@ -157,6 +158,18 @@ class DeviceShieldTrackerActivityViewModel @Inject constructor(
...
+    override fun setDismissed() {
+        vpnStore.dismissNotifyMeInAppTp()
+    }
+
+    override fun isDismissed(): Boolean {
+        return vpnStore.isNotifyMeInAppTpDismissed()
+    }
+
+    override fun visibilityChanged(visible: Boolean) {
+        // noop
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModel.kt:170]: No implementation needed here as there is nothing to do in case the visibility changed. This makes sense e.g when the view is added to a recyclerview.

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt [hunk 4]
```diff
@@ -128,38 +141,38 @@ class DownloadsViewModel @Inject constructor(
     }
 
     fun onQueryTextChange(newText: String) {
-        val filtered = LinkedHashMap<Header, List<Item>>()
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt:131]: Moved as is in a separate function.

File: app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@ExperimentalCoroutinesApi
+@Config(manifest = Config.NONE)
+class NotifyMeViewModelTest {
```
Comment (anikiki) [inline app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt:40]: This should be moved to the UI module.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/themepreview/ui/component/ComponentViewHolder.kt [hunk 6]
```diff
@@ -285,6 +286,21 @@ sealed class ComponentViewHolder(val view: View) : RecyclerView.ViewHolder(view)
             val notifyMeView: NotifyMeView = view.findViewById(R.id.notifyMe)
             notifyMeView.setTitle("Component title")
             notifyMeView.setSubtitle("Component subtitle")
+            notifyMeView.setListener(
+                object : NotifyMeListener {
```
Comment (anikiki) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/themepreview/ui/component/ComponentViewHolder.kt:290]: Dummy implementation as we don't want to grant permissions from the Component Preview.

File: common-ui/src/main/res/values/attrs-notify-me-view.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<resources>
+    <declare-styleable name="NotifyMeView">
+        <attr name="titleText" format="string" />
+        <attr name="subtitleText" format="string" />
```
Comment (malmstein) [inline common-ui/src/main/res/values/attrs-notify-me-view.xml:20]:
```suggestion
        <attr name="secondaryText" />
```

File: app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<resources>
+
+    <!-- Notify Me View on Downloads-->
+    <string name="appTrackingProtectionNotifyMeTitle" translatable="false">Keep an eye on blocked trackers</string>
```
Comment (malmstein) [inline app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml:21]: Why is this here? it should be moved to `donottranslate.xml` in `common-ui`. There's already Notify Me strings there.

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadViewItem.kt [hunk 9]
```diff
@@ -20,6 +20,7 @@ import com.duckduckgo.downloads.api.model.DownloadItem
 
 sealed class DownloadViewItem {
     object Empty : DownloadViewItem()
+    object NotifyMe : DownloadViewItem()
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadViewItem.kt:23]: I think this is what cause the flickering shown in the Ship Review.

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsActivity.kt [hunk 10]
```diff
@@ -116,6 +115,12 @@ class DownloadsActivity : DuckDuckGoActivity() {
         }
     }
 
+    override fun onDestroy() {
+        downloadsAdapter.removeNotifyMeListener()
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsActivity.kt:119]: is this needed?

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt [hunk 11]
```diff
@@ -67,23 +75,28 @@ class DownloadsViewModel @Inject constructor(
...
-            }
-        }
+    private val downloadItems: Flow<List<DownloadViewItem>> = downloadsRepository.getDownloadsAsFlow().map {
+        it.mapToDownloadViewItems()
     }
 
-    fun viewState(): StateFlow<ViewState> {
-        return viewState
-    }
+    private val notifyMeVisibilityChanged = MutableStateFlow(true)
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt:84]: Should this be false to start with? I think you can avoid the flickering that way. At least, if there's flickering, should be because the Component appears and not because it disappears.

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt [hunk 12]
```diff
@@ -67,23 +75,28 @@ class DownloadsViewModel @Inject constructor(
...
+
+    private val filterText = MutableStateFlow("")
+
+    val viewState: StateFlow<ViewState> = combine(
+        flow = downloadItems,
+        flow2 = notifyMeVisibilityChanged,
+        flow3 = filterText,
+    ) { items, visible, filter ->
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt:92]: nit: `visible` doesn't explain the output of `notifyMeVisibilityChanged`.

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt [hunk 13]
```diff
@@ -184,6 +197,22 @@ class DownloadsViewModel @Inject constructor(
         }
     }
 
+    override fun setDismissed() {
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt:200]: ditto about then `Listener` being needed.

File: app/src/main/res/values/donottranslate.xml [hunk 14]
```diff
@@ -27,4 +27,8 @@
 
     <!-- We won't have a translation for this, to be used later. -->
     <string name="trackersOverviewEmpty">No tracking requests were blocked from loading on this page. If a company\'s requests are loaded, it can allow them to profile you.</string>
+
+    <!-- Notify Me View on Downloads-->
+    <string name="downloadsNotifyMeTitle" translatable="false">Find out when downloads are ready</string>
```
Comment (malmstein) [inline app/src/main/res/values/donottranslate.xml:32]: nit: move this to `common-ui`?

File: app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt (hunk 5 above)
Comment (malmstein) [reply app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt:39]: Why not do it now?

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeListener.kt [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.ui.notifyme
+
+interface NotifyMeListener {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeListener.kt:19]: see comment above

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+import com.duckduckgo.mobile.android.ui.view.show
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class NotifyMeView : FrameLayout {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:57]: You can overload the constructor to avoid duplication. 

```suggestion
class NotifyMeView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = R.style.Widget_DuckDuckGo_NotifyMeView,
    ) : FrameLayout (context, attrs, defStyleAttr) {
```

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+    ) : this(
+        context,
+        attrs,
+        R.style.Widget_DuckDuckGo_NotifyMeView,
+    )
+
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyle: Int,
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:73]: you're missing the default style, but overloading the constructor as suggest will solve it.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+
+    private val viewModel by lazy {
+        val factory = NotifyMeViewModel.Factory(findViewTreeSavedStateRegistryOwner()!!)
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, factory)[NotifyMeViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.addObserver(viewModel)
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:105]: 💯

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 19]
```diff
@@ -0,0 +1,227 @@
...
+
+        coroutineScope?.cancel()
+        coroutineScope = null
+    }
+
+    fun setListener(listener: NotifyMeListener?) {
+        this.listener = listener
+    }
+
+    fun setTitle(title: String) {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:140]: If you change the attribute name as suggested, remember to change the API to `setPrimaryText` / `setSecondaryText`

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 20]
```diff
@@ -0,0 +1,227 @@
...
+        binding.notifyMeMessageTitle.text = title
+    }
+
+    fun setSubtitle(subtitle: String) {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:144]: ditto

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 21]
```diff
@@ -0,0 +1,227 @@
...
+        // This gets called only on Android 13+.
+        @SuppressLint("InlinedApi")
+        val settingsIntent = Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS)
+            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            .putExtra(Settings.EXTRA_APP_PACKAGE, context.packageName)
+
+        startActivity(context, settingsIntent, null)
+    }
+
+    private fun showMe() {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:186]: nit: Move to a similar interface as [`SearchBar`](https://github.com/duckduckgo/Android/blob/develop/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt#L27), where the Component handles actions and sends events.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 22]
```diff
@@ -0,0 +1,227 @@
...
+
+    private fun showNotificationsPermissionsPrompt() {
+        if (isAtLeastAndroid13()) {
+            ActivityCompat.requestPermissions(getActivity(), arrayOf(Manifest.permission.POST_NOTIFICATIONS), 1)
+        }
+    }
+
+    @ChecksSdkIntAtLeast(api = Build.VERSION_CODES.TIRAMISU)
+    private fun isAtLeastAndroid13(): Boolean {
+        return Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:213]: use `AppBuildConfig` instead. If you move to `Events`, then the dependency can be added to the `ViewModel` and the `View` has less logic.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class NotifyMeViewModel() : ViewModel(), DefaultLifecycleObserver {
+
+    data class ViewState(
+        val visible: Boolean = false,
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt:40]: you have it as visible false by default here, but in other places it's visible by default. I prefer this version.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class NotifyMeViewModel() : ViewModel(), DefaultLifecycleObserver {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt:37]: This has to be internal, so it's hidden to Activity / Fragment that hosts `NotifyMeView`

```suggestion
internal class NotifyMeViewModel() : ViewModel(), DefaultLifecycleObserver {
```

File: common-ui/src/main/res/values/widgets.xml [hunk 25]
```diff
@@ -526,6 +526,7 @@
         <item name="android:textColor">@color/emptyFavoriteActionTextColor</item>
     </style>
 
-
+    <!-- Notify Me component -->
+    <style name="Widget.DuckDuckGo.NotifyMeView"/>
```
Comment (malmstein) [inline common-ui/src/main/res/values/widgets.xml:530]: this is empty, why do you need it? it can be removed

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt (hunk 19 above)
Comment (anikiki) [reply common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt]: Updated.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt (hunk 20 above)
Comment (anikiki) [reply common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt]: Updated.

File: common-ui/src/main/res/values/widgets.xml (hunk 25 above)
Comment (anikiki) [reply common-ui/src/main/res/values/widgets.xml]: Updated.

File: app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt (hunk 5 above)
Comment (anikiki) [reply app/src/test/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModelTest.kt]: Updated.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt (hunk 24 above)
Comment (anikiki) [reply common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeViewModel.kt]: Updated as I moved the test in `common-ui`. That was the blocker for not making it internal.

File: app/src/main/res/values/donottranslate.xml (hunk 14 above)
Comment (anikiki) [reply app/src/main/res/values/donottranslate.xml:32]: Same as above.
_I don't think this string belongs to the common-ui. The common-ui module should not know how the component is used in other modules._

File: app/src/main/java/com/duckduckgo/app/downloads/DownloadsAdapter.kt [hunk 26]
```diff
@@ -26,23 +26,28 @@ import com.duckduckgo.app.browser.R
...
+import com.duckduckgo.app.downloads.DownloadViewItem.NotifyMe
 import com.duckduckgo.app.global.formatters.data.DataSizeFormatter
 import com.duckduckgo.downloads.store.DownloadStatus.FINISHED
 import com.duckduckgo.mobile.android.databinding.RowTwoLineItemBinding
 import com.duckduckgo.mobile.android.ui.menu.PopupMenu
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeListener
 import com.duckduckgo.mobile.android.ui.view.gone
 import com.duckduckgo.mobile.android.ui.view.show
 import javax.inject.Inject
+import kotlinx.android.synthetic.main.view_item_downloads_notify_me.view.*
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/downloads/DownloadsAdapter.kt:42]: We don't use synthetics anymore, see https://app.asana.com/0/414730916066338/1198731892860153/f

File: app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml (hunk 8 above)
Comment (marcosholgado) [inline app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml:21]: Don't need translatable=false inside donottranslate.xml

Comment (malmstein) [reply app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml]: All your logic / layouts are in the `common-ui` module, why do you want to treat text differently?

File: app/src/main/res/values/donottranslate.xml (hunk 14 above)
Comment (malmstein) [reply app/src/main/res/values/donottranslate.xml:32]: ditto

File: app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml (hunk 8 above)
Comment (anikiki) [reply app-tracking-protection/vpn-impl/src/main/res/values/donottranslate.xml]: `common-ui` defines the logic / layouts and the modules using the components customise them. Those modules are the ones who can decide what strings to display using the component's custom attributes.

The strings from `donottranslate.xml` in `common-ui` are used for UI elements in components which are not customisable. One example is the label for the `Notify me` button which I added in `common-ui`.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModel.kt [hunk 27]
```diff
@@ -51,7 +52,7 @@ class DeviceShieldTrackerActivityViewModel @Inject constructor(
     private val vpnFeatureRemover: VpnFeatureRemover,
     private val vpnStore: VpnStore,
     private val dispatcherProvider: DispatcherProvider,
-) : ViewModel() {
+) : ViewModel(), NotifyMeListener {
```
[the remark this answers is not in the document]
Comment (anikiki) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivityViewModel.kt:55]: The component can be displayed in different screens. If you dismiss from one screen it should still be visible in the others. To implement this logic inside the view it needs to know the context where it is displayed.

I see only one way to do it. 
Add a new `screenName` custom attribute which must be unique and equal to the screen name. Internally the component will use its own shared preferences to track the dismiss state for each key. It will also send the pixel with the `screenName` provided (as per the latest requirement).

The last function in the interface `visibilityChanged` can be replaced by a `onVisibilityChanged(listener)` function defined in the component. This is similar to how Android provides other listeners.

Is this what you had in mind, or something else?
